File: L007/anyBaseToDecimal.cpp
#include<iostream>
#include<Math.h>
using namespace std;


int abtd(int n, int conB){

    int sum=0;
    int p=1;
    while(n != 0){
        int r = n % 10;
        n = n / 10;
        sum += r * p;
        p *= conB;
    }

    return sum;
}

int main(){

    int n,conB;
    cin>>n;

    cin>>conB;
    cout<<abtd(n, conB);
    return 0;
}

File: L007/anyBaseToDecimal.java
import java.util.Scanner;

public class decimalToAnyBase {

    public static Scanner sc = new Scanner(System.in);

    public static int dtab(int n, int a) {

        int sum = 0;
        int p = 1;
        while (n != 0) {
            int r = n % 10;
            n = n / 10;

            sum += r * p;
            p *= a;
        }

        return sum;
    }

    public static void main(String[] args) {

        int n = sc.nextInt();

        int conB = sc.nextInt();

        dtab(n, conB);
    }
}